packages/perseus/src/widgets/expression.tsx

```diff
@@ -66,16 +55,17 @@ type Rubric = PerseusExpressionWidgetOptions;
 
 type ExternalProps = WidgetProps<RenderProps, Rubric>;
 
-export type Props = ExternalProps & {
-    apiOptions: NonNullable<ExternalProps["apiOptions"]>;
-    buttonSets: NonNullable<ExternalProps["buttonSets"]>;
-    functions: NonNullable<ExternalProps["functions"]>;
-    linterContext: NonNullable<ExternalProps["linterContext"]>;
-    onBlur: NonNullable<ExternalProps["onBlur"]>;
-    onFocus: NonNullable<ExternalProps["onFocus"]>;
-    times: NonNullable<ExternalProps["times"]>;
-    value: string;
-};
+export type Props = ExternalProps &
+    ReturnType<typeof useDependencies> & {
```

Comment:
Just of curiosity, why are we using `typeof` here instead of defining a type for `useDependencies` and importing it with it the function?

Reply:
I'm using `ReturnType<>` to extract the type returned by `useDepencies`. `ReturnType<>` takes a type so I need to use `typeof` to get the type of the `useDependencies` function.  In my mind, this was cleaner than declaring and exporting the return type of `useDependencies`. It seemed like less to keep in sync and TypeScript has rich utility types to do what I needed, so I went with that approach.